ingestion/src/metadata/profiler/metrics/static/distinct_count.py

```diff
@@ -43,15 +43,16 @@ def fn(self):
         return func.count(distinct(column(self.col.name)))
 
     def df_fn(self, dfs=None):
-        try:
-            import pandas as pd  # pylint: disable=import-outside-toplevel
+        from collections import Counter  # pylint: disable=import-outside-toplevel
 
-            return len(
-                pd.concat((df[self.col.name] for df in dfs)).dropna().drop_duplicates()
-            )
+        try:
+            counter = Counter()
+            for df in dfs:
+                counter.update(df[self.col.name].dropna().to_list())
+            return len(counter.keys())
```

Comment:
From what I am seeing `.value_counts().to_dict()` is about 20% faster (https://github.com/open-metadata/OpenMetadata/pull/10603#discussion_r1157101575)